public/app/core/components/RolePicker/TeamRolePicker.tsx

```diff
@@ -13,23 +13,30 @@ export interface Props {
 }
 
 export const TeamRolePicker: FC<Props> = ({ teamId, orgId, roleOptions, disabled, builtinRolesDisabled }) => {
-  const [appliedRoles, setAppliedRoles] = useState<Role[]>([]);
-
-  const { loading } = useAsync(async () => {
+  const [{ loading, value: appliedRoles }, getTeamRoles] = useAsyncFn(async () => {
     try {
-      const teamRoles = await fetchTeamRoles(teamId, orgId);
-      setAppliedRoles(teamRoles);
+      return await fetchTeamRoles(teamId, orgId);
     } catch (e) {
       // TODO handle error
       console.error('Error loading options');
     }
+    return [];
   }, [orgId, teamId]);
 
+  useEffect(() => {
+    getTeamRoles();
+  }, [orgId, teamId, getTeamRoles]);
+
+  const onRolesChange = async (roles: string[]) => {
+    await updateTeamRoles(roles, teamId, orgId);
+    await getTeamRoles();
+  };
+
   return (
     <RolePicker
-      onRolesChange={(roles) => updateTeamRoles(roles, teamId, orgId)}
+      onRolesChange={onRolesChange}
       roleOptions={roleOptions}
-      appliedRoles={appliedRoles}
+      appliedRoles={appliedRoles || []}
```

Comment:
The default value could be also provided via the third arg to `useAsyncFn`, i.e. `{value: []}` or passed during destructuring `const [{ loading, value: appliedRoles = [] }, getTeamRoles]` to make the render a bit cleaner.